src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -391,24 +360,101 @@ private void LowerBasicSwitch(DecisionTree.ByValue byValue)
                 }
```

Comment:
I'll place the assertion into the switch, below.

---
In reply to: [70700915](https://github.com/dotnet/roslyn/pull/12478#discussion_r70700915) [](ancestors = 70700915)